homeassistant/components/sensibo/coordinator.py

```diff
@@ -0,0 +1,120 @@
+"""DataUpdateCoordinator for the Sensibo integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+from typing import Any
+
+from aiohttp.client_exceptions import ClientConnectionError
+import pysensibo
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, FIELD_TO_FLAG, LOGGER, TIMEOUT
+
+
+class SensiboDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Sensibo Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Sensibo coordinator."""
+        self.client = pysensibo.SensiboClient(
+            entry.data[CONF_API_KEY],
+            session=async_get_clientsession(hass),
+            timeout=TIMEOUT,
+        )
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[str, dict[str, Any]]:
+        """Fetch data from Sensibo."""
+
+        devices = []
+        try:
+            for dev in await self.client.async_get_devices():
+                devices.append(dev)
+        except (
+            ClientConnectionError,
+            asyncio.TimeoutError,
+            pysensibo.SensiboError,
+        ) as error:
+            raise UpdateFailed from error
+
+        entities: dict[str, dict[str, Any]] = {}
```

Comment:
This name is misleading. The coordinator shouldn't know about entities. It's ok to structure the data like this, but I'd call the variable something else. It's still device data, just restructured.